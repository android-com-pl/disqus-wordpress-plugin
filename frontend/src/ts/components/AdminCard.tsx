import * as React from 'react';

export interface IAdminCardProps {
    children?: any;
    title: string;
}

const AdminCard = (props: IAdminCardProps) => (
    <div className='card dsq-card'>
        {props.title ? <h2 className='title'>{props.title}</h2> : null}
        {props.children}
    </div>
);

export default AdminCard;
