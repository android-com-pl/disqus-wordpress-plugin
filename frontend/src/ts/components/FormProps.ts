import * as React from 'react';
import AdminState from '../reducers/AdminState';

export interface IFormProps {
    data: AdminState;
    onCopyText(textValue: string): void;
    onToggleState(stateKey: string): void;
    onInputChange(stateKey: string, event: React.SyntheticEvent<HTMLInputElement>): void;
    onSubmitSiteForm(event: React.SyntheticEvent<HTMLFormElement>): void;
    onSubmitSyncConfigForm(event: React.SyntheticEvent<HTMLFormElement>): void;
}
