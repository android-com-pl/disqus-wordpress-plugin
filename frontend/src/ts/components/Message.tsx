import * as React from 'react';
import { IMessage } from '../reducers/AdminState';

const getDismissButton = (props: IMessage) => {
    if (!props.onDismiss)
        return null;

    return (
        <button type="button" className="notice-dismiss" onClick={props.onDismiss}>
            <span className="screen-reader-text">Dismiss this notice.</span>
        </button>
    );
};

const Message = (props: IMessage) => (
    <div className={`notice notice-${props.type} inline is-dismissible`}>
        <p>{props.text}</p>
        {getDismissButton(props)}
    </div>
);

export default Message;
