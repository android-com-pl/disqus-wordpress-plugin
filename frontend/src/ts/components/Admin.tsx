import * as React from 'react';
import * as ReactRedux from 'react-redux';
import * as Redux from 'redux';
import {
    SiteConfigContainer,
    SSOConfigContainer,
    SyncConfigContainer,
} from '../containers';
import { IAdminState } from '../reducers/AdminState';
import { getForumAdminUrl, getWordpressAdminUrl } from '../utils';
import AdminCard from './AdminCard';
import { IFormProps } from './FormProps';
import WelcomePanel from './WelcomePanel';

const getSSOContainer = (props: IFormProps) => {
    const adminOptions = props.data.adminOptions;
    if (!adminOptions.disqus_public_key || !adminOptions.disqus_secret_key) {
        return (
            <div className="notice notice-warning">
                <p>
                    <span className="dashicons dashicons-warning" />
                    {' '}
                    {__('You must have an API Public Key and API Secret Key configured to enable this feature.')}
                </p>
            </div>
        );
    }
    return <SSOConfigContainer />;
};

const getSyncContainer = (props: IFormProps) => {
    const adminOptions = props.data.adminOptions;
    if (!adminOptions.disqus_secret_key || !adminOptions.disqus_admin_access_token) {
        return (
            <div className="notice notice-warning">
                <p>
                    <span className="dashicons dashicons-warning" />
                    {' '}
                    {__('You must have an API Secret Key and API Access Token configured to enable this feature.')}
                </p>
            </div>
        );
    }

    return <SyncConfigContainer />;
};

/* tslint:disable:max-line-length */
const Admin = (props: IFormProps) => (
    <div>
        <WelcomePanel shortname={props.data.adminOptions.disqus_forum_url} />
        <h2 className="title">
            {__('Plugin Settings')}
        </h2>
        <div>
            <AdminCard title={__('Site Configuration')}>
                <p className="description">
                    {__('Your site configuration comes from Disqus. Changing these values may break your installation.')}
                    {' '}
                    <a
                        href={getForumAdminUrl(props.data.adminOptions.disqus_forum_url, 'install/platforms/wordpress')}
                        target="_blank"
                    >
                        {__('WordPress install instructions')}
                    </a>
                </p>
                <SiteConfigContainer />
            </AdminCard>
            <AdminCard title={__('Single Sign-on')}>
                <p className="description">
                    {__('Allow users to sign in with this site\'s user accounts. This is a premium-level feature and must be enabled for your organization.')}
                    {' '}
                    <a href="https://disqus.com/api/applications/" target="_blank">
                        {__('Learn More')}
                    </a>
                </p>
                {getSSOContainer(props)}
            </AdminCard>
            <AdminCard title={__('WordPress Comments')}>
                <p className="description">
                    {__('Disqus has replaced the default WordPress commenting system. You may access and edit the comments in your database, but any actions performed there will not be reflected in Disqus.')}
                </p>
                <p className="submit">
                    <a
                        href={getWordpressAdminUrl('editComments')}
                        className="button"
                    >
                        {__('View WordPress Comments')}
                    </a>
                </p>
                <hr />
                <h3>
                    {__('Syncing')}
                </h3>
                <p className="description">
                    {__('Syncing will copy comments created and edited in Disqus to your local WordPress database for backup purposes. This will create additional work for your database/server and is not recommended for busy communities.')}
                </p>
                {getSyncContainer(props)}
            </AdminCard>
        </div>
    </div>
);
/* tslint:enable:max-line-length */

export default Admin;
