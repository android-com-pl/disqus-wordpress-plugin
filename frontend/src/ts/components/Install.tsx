import * as React from 'react';
import * as ReactRedux from 'react-redux';
import { SiteConfigContainer } from '../containers';
import { IAdminState } from '../reducers/AdminState';
import { IDisqusWordpressWindow } from '../reducers/AdminState';
import { appendQueryToUrl } from '../utils';
import { getWordpressAdminUrl } from '../utils';
import AdminCard from './AdminCard';
import { IFormProps } from './FormProps';

const WIN = window as IDisqusWordpressWindow;
const REST_OPTIONS = WIN.DISQUS_WP.rest;

/* tslint:disable:max-line-length */
const Install = (props: IFormProps) => (
    <div>
        <AdminCard title={__('Automatic Installation')}>
            <p>
                {__('Start the registration process and choose the WordPress platform when prompted.')}
            </p>
            <table className="form-table">
                <tbody>
                    <tr>
                        <th scope="row">
                            <label htmlFor="configKey">{__('Install Key')}</label>
                        </th>
                        <td>
                            <input
                                id="configKey"
                                type="text"
                                value={appendQueryToUrl(`${REST_OPTIONS.base}settings`, `secret=${props.data.adminOptions.get('disqus_sync_token', '')}`)}
                                className="regular-text"
                                readOnly={true}
                            />
                            <p className="description">
                                {__('Copy and paste this key in the WordPress installation instructions.')}
                            </p>
                        </td>
                    </tr>
                </tbody>
            </table>
            <p className="submit">
                <a
                    className="button button-primary"
                    href="https://disqus.com/profile/login/?next=/admin/create/"
                    target="_blank"
                >
                    {__('Start Installation')}
                </a>
                <br />
                <br />
                <em>
                    {__('Already have a site registered?')}
                    {' '}
                    <a href="https://disqus.com/admin/install/platforms/wordpress/">
                        Go to WordPress installation page
                    </a>.
                </em>
            </p>
        </AdminCard>
        <AdminCard title={__('Manual Installation')}>
            <p>
                {__('You may install Disqus manually if you\'re not able to use the automatic installer.')}
            </p>
            <p className="submit">
                <button className="button button-link" onClick={props.onToggleState.bind(null, 'isSiteFormLocked')}>
                    <span className={`dashicons dashicons-arrow-${props.data.isSiteFormLocked ? 'right' : 'down'}`} />
                    {' '}
                    {props.data.isSiteFormLocked ? __('Show manual configuration') : __('Hide manual configuration')}
                </button>
            </p>
            {props.data.isSiteFormLocked ? null : <SiteConfigContainer />}
        </AdminCard>
        <AdminCard title={__('WordPress Comments')}>
            <p className="description">
                {__('Disqus has replaced the default WordPress commenting system. You may access and edit the comments in your database, but any actions performed there will not be reflected in Disqus.')}
            </p>
            <p className="submit">
                <a
                    href={getWordpressAdminUrl('editComments')}
                    className="button"
                >
                    {__('View WordPress Comments')}
                </a>
            </p>
        </AdminCard>
    </div>
);
/* tslint:enable:max-line-length */

export default Install;
