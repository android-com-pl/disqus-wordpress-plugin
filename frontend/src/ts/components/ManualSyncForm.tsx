import * as moment from 'moment';
import * as React from 'react';
import SyncStatus from '../reducers/SyncStatus';
import { IFormProps } from './FormProps';

const ManualSyncForm = (props: IFormProps) => {
    return (
        <form
            name='manual-sync'
            method='POST'
            onSubmit={props.onSubmitManualSyncForm}
        >
            <h4>{__('Manually Sync Comments')}</h4>
            <p className='description'>
                {__('Select a time range to sync past comments. Date ranges are limited to a maximum of 12 months.')}
            </p>
            <table className='form-table'>
                <tbody>
                    <tr>
                        <th scope='row'>
                            <label htmlFor='manualSyncRangeStart'>
                                {__('Start Date')}
                            </label>
                        </th>
                        <td>
                            <input
                                type='date'
                                id='manualSyncRangeStart'
                                name='manualSyncRangeStart'
                                className='regular-text'
                                value={props.data.manualSyncRangeStart}
                                onChange={props.onDateSelectorInputchange.bind(null, 'manualSyncRangeStart')}
                                max={props.data.manualSyncRangeEnd}
                                min={moment(props.data.manualSyncRangeEnd).subtract(12, 'months').format('YYYY-MM-DD')}
                                disabled={props.data.isManualSyncRunning}
                            />
                            <p className='description'>
                                {__('The start date for the manual sync')}
                            </p>
                        </td>
                    </tr>
                    <tr>
                        <th scope='row'>
                            <label htmlFor='manualSyncRangeEnd'>
                                {__('End Date')}
                            </label>
                        </th>
                        <td>
                            <input
                                type='date'
                                id='manualSyncRangeEnd'
                                name='manualSyncRangeEnd'
                                className='regular-text'
                                value={props.data.manualSyncRangeEnd}
                                onChange={props.onDateSelectorInputchange.bind(null, 'manualSyncRangeEnd')}
                                max={moment().format('YYYY-MM-DD')}
                                min={props.data.manualSyncRangeStart}
                                disabled={props.data.isManualSyncRunning}
                            />
                            <p className='description'>
                                {__('The end date for the manual sync')}
                            </p>
                        </td>
                    </tr>
                </tbody>
            </table>
            <p className='submit'>
                <button type='submit' className='button button-large' disabled={props.data.isManualSyncRunning}>
                    <span className='dashicons dashicons-update' />
                    {' '}
                    {__('Run Manual Sync')}
                </button>
            </p>
        </form>
    );
};

export default ManualSyncForm;
