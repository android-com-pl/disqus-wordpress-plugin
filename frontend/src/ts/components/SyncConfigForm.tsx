import * as React from 'react';
import SyncStatus from '../reducers/SyncStatus';
import { IFormProps } from './FormProps';

const getSyncStatus = (syncStatus: SyncStatus) => {
    if (!syncStatus.subscribed) {
        return {
            button: __('Enable Auto Syncing'),
            endpoint: 'sync/enable',
            status: __('Not Configured'),
            statusIcon: 'play',
        };
    }

    if (syncStatus.requires_update) {
        return {
            button: __('Update Auto Syncing'),
            endpoint: 'sync/enable',
            status: __('Needs Update'),
            statusIcon: 'play',
        };
    }

    if (syncStatus.enabled) {
        return {
            button: __('Pause Auto Syncing'),
            endpoint: 'sync/disable',
            status: __('Running'),
            statusIcon: 'pause',
        };
    }

    return {
        button: __('Enable Auto Syncing'),
        endpoint: 'sync/enable',
        status: __('Paused'),
        statusIcon: 'play',
    };
};

const SyncConfigForm = (props: IFormProps) => {
    const syncStatus = getSyncStatus(props.data.syncStatus);
    return (
        <form
            name={syncStatus.endpoint}
            method='POST'
            onSubmit={props.onSubmitSyncConfigForm}
        >
            <p>
                <span className='dashicons dashicons-update' />
                {' '}
                {__('Auto syncing')}{':'}
                {' '}
                <strong>{syncStatus.status}</strong>
            </p>
            {props.data.syncStatus.last_message ? <p>{props.data.syncStatus.last_message}</p> : null}
            <p className='submit'>
                <button type='submit' className='button button-large'>
                    <span className={`dashicons dashicons-controls-${syncStatus.statusIcon}`} />
                    {' '}
                    {syncStatus.button}
                </button>
            </p>
        </form>
    );
};

export default SyncConfigForm;
