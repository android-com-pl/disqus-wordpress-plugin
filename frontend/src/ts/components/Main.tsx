import * as React from 'react';
import InstallContainer from '../containers/InstallContainer';
import SyncTokenContainer from '../containers/SyncTokenContainer';
import { getWordpressAdminUrl } from '../utils';
import Admin from './Admin';
import { IFormProps } from './FormProps';
import Loading from './Loading';
import Message from './Message';

const PRE_RELEASE_TYPES: any = Object.freeze({
    alpha: 'alpha',
    beta: 'beta',
});

/* tslint:disable:max-line-length */
const getMainView = (props: IFormProps) => {
    if (!props.data.config.permissions.canManageSettings)
        return __('You don\'t have permission to make any changes here. Please contact the site administrator to get access.');
    else if (props.data.isFetchingAdminOptions || props.data.isFetchingSyncStatus)
        return <Loading />;
    else if (!props.data.adminOptions.disqus_sync_token)
        return (
            <div>
                <div className={`notice notice-info inline`}>
                    <p>
                        {__('The plugin was unable to generate a secret key for your site. You will have to create one below in order for installation and syncing to work.')}
                    </p>
                </div>
                <SyncTokenContainer {...props} />
            </div>
        );
    return <Admin {...props} />;
};

const getPreReleaseNotice = (pluginVersion: string) => {
    // Format of versions can be either 1.0.0, 1.0.0-beta, or 1.0.0-beta.1
    const preReleaseType = (pluginVersion.split('-')[1] || '').split('.')[0];
    if (PRE_RELEASE_TYPES[preReleaseType]) {
        return (
            <div className={`notice notice-info inline`}>
                <p>
                    You are using a <strong>pre-release version ({`${pluginVersion}`})</strong> of the Disqus WordPress plugin.
                    {' '}<a href='https://github.com/disqus/disqus-wordpress-plugin/releases' target='_blank'>Check for new releases</a>
                </p>
            </div>
        );
    }
    return null;
};

const Main = (props: IFormProps) => (
    <div className='dsq-admin-wrapper'>
        <div className='wrap'>
            <a href={getWordpressAdminUrl('disqus')} className='disqus-logo'>
                <img
                    src='https://a.disquscdn.com/dotcom/d-2407bda/img/brand/disqus-logo-blue-white.svg'
                    width={102}
                />
            </a>
        </div>
        <div className='wrap'>
            {getPreReleaseNotice(props.data.config.site.pluginVersion)}
            {props.data.message ? <Message {...props.data.message} /> : null}
            {getMainView(props)}
        </div>
    </div>
);
/* tslint:enable:max-line-length */

export default Main;
