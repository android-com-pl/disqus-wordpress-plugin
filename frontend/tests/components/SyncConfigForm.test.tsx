import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { IFormProps } from '../../src/ts/components/FormProps';
import SyncConfigForm from '../../src/ts/components/SyncConfigForm';
import { getDefaultFormProps } from '../fixtures';

describe('SyncConfigForm component', () => {
    test('Renders form with no sync enabled', () => {
        const props: IFormProps = getDefaultFormProps();
        const component = TestRenderer.create(<SyncConfigForm {...props} />);

        expect(component.toJSON()).toMatchSnapshot();
    });

    test('Renders form with disabled sync subscription', () => {
        const props: IFormProps = getDefaultFormProps();
        props.data = props.data.with({
            syncStatus: props.data.syncStatus.with({
                subscribed: false,
            }),
        });
        const component = TestRenderer.create(<SyncConfigForm {...props} />);

        expect(component.toJSON()).toMatchSnapshot();
    });

    test('Renders form with paused sync subscription', () => {
        const props: IFormProps = getDefaultFormProps();
        props.data = props.data.with({
            syncStatus: props.data.syncStatus.with({
                requires_update: true,
                subscribed: true,
            }),
        });
        const component = TestRenderer.create(<SyncConfigForm {...props} />);

        expect(component.toJSON()).toMatchSnapshot();
    });

    test('Renders form with enabled sync subscription', () => {
        const props: IFormProps = getDefaultFormProps();
        props.data = props.data.with({
            syncStatus: props.data.syncStatus.with({
                enabled: true,
                subscribed: true,
            }),
        });
        const component = TestRenderer.create(<SyncConfigForm {...props} />);

        expect(component.toJSON()).toMatchSnapshot();
    });

    test('Submitting the form calls onSubmitSyncConfigForm prop', () => {
        const props: IFormProps = getDefaultFormProps();
        const onSubmitSyncConfigForm = props.onSubmitSyncConfigForm = jest.fn();
        props.data = props.data.with({
            syncStatus: props.data.syncStatus.with({
                subscribed: false,
            }),
        });
        const component = TestRenderer.create(<SyncConfigForm {...props} />);
        const root = component.root;

        const form = root.findByType('form');
        form.props.onSubmit();

        expect(onSubmitSyncConfigForm).toHaveBeenCalled();
    });
});
