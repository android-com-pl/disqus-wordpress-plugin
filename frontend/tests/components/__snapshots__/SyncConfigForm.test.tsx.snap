// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SyncConfigForm component Renders form with disabled sync subscription 1`] = `
<form
  method="POST"
  name="sync/enable"
  onSubmit={[MockFunction]}
>
  <p>
    <span
      className="dashicons dashicons-update"
    />
     
    Auto syncing
    :
     
    <strong>
      Not Configured
    </strong>
  </p>
  <p
    className="submit"
  >
    <button
      className="button button-large"
      type="submit"
    >
      <span
        className="dashicons dashicons-controls-play"
      />
       
      Enable Auto Syncing
    </button>
  </p>
</form>
`;

exports[`SyncConfigForm component Renders form with enabled sync subscription 1`] = `
<form
  method="POST"
  name="sync/disable"
  onSubmit={[MockFunction]}
>
  <p>
    <span
      className="dashicons dashicons-update"
    />
     
    Auto syncing
    :
     
    <strong>
      Running
    </strong>
  </p>
  <p
    className="submit"
  >
    <button
      className="button button-large"
      type="submit"
    >
      <span
        className="dashicons dashicons-controls-pause"
      />
       
      Pause Auto Syncing
    </button>
  </p>
</form>
`;

exports[`SyncConfigForm component Renders form with no sync enabled 1`] = `
<form
  method="POST"
  name="sync/enable"
  onSubmit={[MockFunction]}
>
  <p>
    <span
      className="dashicons dashicons-update"
    />
     
    Auto syncing
    :
     
    <strong>
      Not Configured
    </strong>
  </p>
  <p
    className="submit"
  >
    <button
      className="button button-large"
      type="submit"
    >
      <span
        className="dashicons dashicons-controls-play"
      />
       
      Enable Auto Syncing
    </button>
  </p>
</form>
`;

exports[`SyncConfigForm component Renders form with paused sync subscription 1`] = `
<form
  method="POST"
  name="sync/enable"
  onSubmit={[MockFunction]}
>
  <p>
    <span
      className="dashicons dashicons-update"
    />
     
    Auto syncing
    :
     
    <strong>
      Needs Update
    </strong>
  </p>
  <p
    className="submit"
  >
    <button
      className="button button-large"
      type="submit"
    >
      <span
        className="dashicons dashicons-controls-play"
      />
       
      Update Auto Syncing
    </button>
  </p>
</form>
`;
